java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1464 @@
+    lastValueAllocationSizeInBytes = (int) size;
+    lastValueCapacity = valueCount;
+  }
+
+  /**
+   * Sets the desired value capacity for the vector. This function doesn't
+   * allocate any memory for the vector.
+   *
+   * @param valueCount desired number of elements in the vector
+   * @param density average number of bytes per variable width view element
+   */
+  @Override
+  public void setInitialCapacity(int valueCount, double density) {
+    final long size = (long) valueCount * VIEW_BUFFER_SIZE;
+    initialDataBufferSize = (int) (valueCount * density);
+    checkDataBufferSize(size);
+    lastValueAllocationSizeInBytes = (int) size;
+    lastValueCapacity = valueCount;
+  }
+
+  /**
+   * Get the density of this ListVector.
+   * @return density
+   */
+  public double getDensity() {
+    if (valueCount == 0) {
+      return 0.0D;
+    }
+    final double totalListSize = getTotalLengthUptoIndex(valueCount);
+    return totalListSize / valueCount;
+  }
+
+  /**
+   * Get the current capacity which does not exceed either validity buffer or value buffer.
+   * Note: Here the `getValueCapacity` has a relationship with the value buffer.
+   *
+   * @return number of elements that vector can hold.
+   */
+  @Override
+  public int getValueCapacity() {
+    final int validityCapacity = getValidityBufferValueCapacity();
+    final int valueBufferCapacity = Math.max(capAtMaxInt(viewBuffer.capacity() / VIEW_BUFFER_SIZE), 0);
+    return Math.min(valueBufferCapacity, validityCapacity);
+  }
+
+  private int getValidityBufferValueCapacity() {
+    return capAtMaxInt(validityBuffer.capacity() * 8);
+  }
+
+  /**
+   * zero out the vector and the data in associated buffers.
+   */
+  public void zeroVector() {
+    initValidityBuffer();
+    viewBuffer.setZero(0, viewBuffer.capacity());
+  }
+
+  /* zero out the validity buffer */
+  private void initValidityBuffer() {
+    validityBuffer.setZero(0, validityBuffer.capacity());
+  }
+
+  /**
+   * Reset the vector to initial state. Same as {@link #zeroVector()}.
+   * Note that this method doesn't release any memory.
+   */
+  @Override
+  public void reset() {
+    zeroVector();
+    lastSet = -1;
+    valueCount = 0;
+  }
+
+  /**
+   * Close the vector and release the associated buffers.
+   */
+  @Override
+  public void close() {
+    clear();
+  }
+
+  /**
+   * Same as {@link #close()}.
+   */
+  @Override
+  public void clear() {
+    validityBuffer = releaseBuffer(validityBuffer);
+    viewBuffer = releaseBuffer(viewBuffer);
+    clearDataBuffers();
+    lastSet = -1;
+    valueCount = 0;
+  }
+
+  /**
+  * Release the data buffers and clear the list.
+  */
+  public void clearDataBuffers() {
+    for (ArrowBuf buffer : dataBuffers) {
+      buffer.getReferenceManager().release();
+    }
+    dataBuffers.clear();
+  }
+
+  /**
+   * Get the inner vectors.
+   *
+   * @deprecated This API will be removed as the current implementations no longer support inner vectors.
```

Comment:
Sure I can do that.

Reply:
@lidavidm we have two general filings 
1. https://github.com/apache/arrow/issues/15167
2. https://github.com/apache/arrow/issues/33022

I think [1] suits better. I can list this one there, but it would be better if we do the investigation and list all deprecated warnings and remove them in a graceful manner.

Reply:
Issue for this: https://github.com/apache/arrow/issues/41069